nibabel/tests/test_quaternions.py

```diff
@@ -69,6 +69,20 @@ def test_fillpos():
     assert wxyz[0] == 0.0
 
 
+@pytest.mark.parametrize('dtype', ('f4', 'f8'))
+def test_fillpos_precision(dtype):
```

Comment:
Is the idea that one of these 50 random vectors will generate a small negative difference from zero?  Worth testing that explicitly?

Reply:
Would this be easier to read with some example of a positive and negative error?  Here we are leaving it to chance whether the test fails, are we not?

Reply:
Just to be clear, you're suggesting two vectors that are exemplars of positive and negative error? Would random generation work, as long as we verify that we get them?

```Python
def make_unit_vector(shape, dtype, constraint=lambda x: True, tries=50):
    for _ in range(tries):
        vec = rand.uniform(low=-1, high=1, size=shape).astype(dtype)
        vec /= np.sqrt(vec @ vec)
        if constraint(vec):
            return vec

short = make_unit_vector((3,), dtype, lambda x: x @ x < 1)
long = make_unit_vector((3,), dtype, lambda x: x @ x > 1)

assert nq.fillpositive(short, w2_thresh)[0] == 0
assert nq.fillpositive(long, w2_thresh)[0] == 0
```

Reply:
It seems to me cleaner to find some examplar vectors that we know give positive and negative < threshold differences, and a > threshold difference, and then test some random vectors - rather than hoping (even with reasonable expectation) that we'll get suitable vectors.